### DISCUSSION TITLE
Security

### DISCUSSION THREAD

**Discussion by hhinma** [aessing/udm-backup-ftp#12] [General]
I do not know enough about docker containers to understand this.  Does this implementation use sftp, or ftp over ssh, or is is plain old insecure ftp?

**Comment by aessing** [comment]
Hi,

I configured my FTP server to enforce SFTP connections. The client in the container (LFTP) recognizes this and connects encrypted to the FTP server and copies the backup.
The container doesn't use SSH.

In general, this is an amazingly simple container. The container itself is only a Linux with LFTP installed. The magic is done by the docker container mounts and the BASH script that runs inside the container.

Cheers
Andre

**Reply by hhinma** [reply]
ok, by looking at the network traffic,  it seems that the container does not use SFTP, it uses SSL over FTP, which explains why it cannot connect to my SFTP server unless of course my SFTP/FTP server configuration options are mislabeled.  So, to summarise,  with just SFTP enabled, the connection is rejected and the operation fails.  With SSL over FTP enabled, the connection succeeds.

**Reply by aessing** [reply]
Oh sorry, that was my fault. I mixed up protocols. 

The container does real FTP, which means it uses FTPS, which means FTP with SSL encryption (not SFTP). Here I mixed up the services. The container can not use FTP over SSH (SFTP).
